# $DragonFly: src/sys/dev/acpica5/tables,tbxface.c.patch,v 1.1 2007/01/19 16:41:28 y0netan1 Exp $

--- tbxface.c.orig	2007-01-18 01:29:44.000000000 +0900
+++ tbxface.c	2007-01-20 01:09:28.000000000 +0900
@@ -562,13 +562,16 @@
                 AcpiGbl_RootTableList.Tables[ACPI_TABLE_INDEX_DSDT].Pointer, &Table);
     if (ACPI_SUCCESS (Status) && Table)
     {
+	ACPI_TABLE_DESC *dsdt;
+
         /*
          * DSDT table has been found
          */
-        AcpiTbDeleteTable (ACPI_TABLE_INDEX_DSDT);
-        AcpiGbl_RootTableList.Tables[ACPI_TABLE_INDEX_DSDT].Pointer = Table;
-        AcpiGbl_RootTableList.Tables[ACPI_TABLE_INDEX_DSDT].Length = Table->Length;
-        AcpiGbl_RootTableList.Tables[ACPI_TABLE_INDEX_DSDT].Flags = ACPI_TABLE_ORIGIN_UNKNOWN;
+        dsdt = &AcpiGbl_RootTableList.Tables[ACPI_TABLE_INDEX_DSDT];
+        AcpiTbDeleteTable (dsdt);
+        dsdt->Pointer = Table;
+        dsdt->Length = Table->Length;
+        dsdt->Flags = ACPI_TABLE_ORIGIN_UNKNOWN;
 
         ACPI_INFO ((AE_INFO, "Table DSDT replaced by host OS"));
         AcpiTbPrintTableHeader (0, Table);
